### PR TITLE
Http interface tests

### PR DESCRIPTION
**Pull request by JL-Brothers**
Closes issue #833

### CODE REVIEW

File: openc3/spec/interfaces/http_client_interface_hal_spec.rb [hunk 1]
```diff
@@ -0,0 +1,151 @@
+
```
Comment (jmthomas) [inline openc3/spec/interfaces/http_client_interface_hal_spec.rb:1]: What is the `hal` in the file name for? We already have a `http_client_interface_spec.rb`. These additional methods should just go in there.

File: openc3/spec/interfaces/http_server_interface_hal_spec.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
+require 'spec_helper'
```
Comment (jmthomas) [inline openc3/spec/interfaces/http_server_interface_hal_spec.rb:1]: What is the `hal` in the file name for? We already have a `http_server_interface_spec.rb`. These additional methods should just go in there.

**Review by jmthomas — CHANGES_REQUESTED** [review]

File: openc3/spec/interfaces/http_client_interface_hal_spec.rb (hunk 1 above)
Comment (JL-Brothers) [reply openc3/spec/interfaces/http_client_interface_hal_spec.rb]: Yes, done.

File: openc3/spec/interfaces/http_server_interface_hal_spec.rb (hunk 2 above)
Comment (JL-Brothers) [reply openc3/spec/interfaces/http_server_interface_hal_spec.rb]: WIP.  The hal file is gone.

File: openc3/spec/interfaces/http_client_interface_spec.rb [hunk 3]
```diff
@@ -42,7 +58,225 @@ module OpenC3
...
+      @interface = HttpClientInterface.new('example.com', 8080, 'https', 10, 15, 5, true)
+    end
+
+    describe "#initialize" do
+      it "initializes with default parameters" do
+        interface = HttpClientInterface.new('example.com')
+        expect(interface.instance_variable_get(:@hostname)).to eq('example.com')
+        expect(interface.instance_variable_get(:@port)).to eq(80)
+        expect(interface.instance_variable_get(:@protocol)).to eq('http')
+      end
```
Comment (jmthomas) [inline openc3/spec/interfaces/http_client_interface_spec.rb:73]: Combine with the existing ... check for i.name

File: openc3/spec/interfaces/http_client_interface_spec.rb [hunk 4]
```diff
@@ -42,7 +58,225 @@ module OpenC3
...
+        expect(@interface.instance_variable_get(:@read_timeout)).to eq(15.0)
+        expect(@interface.instance_variable_get(:@connect_timeout)).to eq(5.0)
+        expect(@interface.instance_variable_get(:@include_request_in_response)).to be true
+      end
+    end
+
+    describe "#connection_string" do
+      it "returns the correct URL" do
+        expect(@interface.connection_string).to eq('https://example.com:8080')
+      end
```
Comment (jmthomas) [inline openc3/spec/interfaces/http_client_interface_spec.rb:89]: Combine with existing ... the existing tests are more thorough

File: openc3/spec/interfaces/http_client_interface_spec.rb (hunk 3 above)
Comment (JL-Brothers) [reply openc3/spec/interfaces/http_client_interface_spec.rb:57]: Done

File: openc3/spec/interfaces/http_client_interface_spec.rb (hunk 4 above)
Comment (JL-Brothers) [reply openc3/spec/interfaces/http_client_interface_spec.rb:73]: Yes, done now.

**Review by jmthomas — APPROVED** [review]

**Pull request merged**